### DISCUSSION TITLE
Pole intensities vs angular co-ordinates plot

### DISCUSSION THREAD

**Discussion by rrkamath** [mtex-toolbox/mtex#554] [Ask Anything] [answered]
Hi all,

I wanted to plot the intensity of a pole figure versus the angular co-ordinates to plot the upper hemisphere as a "normal graph" (using plot3d) instead of it being a hemisphere. 

Could you please help me in this regard ? Thanks !

Best
Rakesh

**Comment by ralfHielscher** [comment]
I do not really understand how you want to plot, but maybe
```matlab
plot(pf,'3d')
```
or
```matlab
plot(pf,'contourf')
```
give you what you want.

Ralf.

**Reply by rrkamath** [reply]
Hi Ralf,

My bad, i'll try to put it in a different way. I wanted to extract the angular co-ordinates and corresponding pole intensities from the pf so that I can plot pole intensities v/s angular co-ordinates in rectangular co-ordinates (unlike the polar coordinates in a pf). How can I do that ?

Rakesh

**Comment by ralfHielscher** [comment, marked as answer]
How about this:
```matlab
theta = pf.r.theta ./ degree;
rho = pf.r.rho ./ degree;

scatter(theta(:), rho(:), 20, pf.intensities(:))
```
Ralf.

**Comment by rrkamath** [comment]
I tried the same and got the following error:

```
Error using scatter (line 110)
CData must be an RGB triplet, an M-by-1 vector of the same length as X, or an M-by-3 matrix.

Error in L1_9_BCC (line 136)
scatter(theta(:), rho(:), 20, pf.intensities)
```
I also noticed that pf.r.rho and pf.r.theta (1368X11) have different dimensions compared to pf.intensities (18X836). Do you think that is the problem ?

Rakesh

**Reply by ralfHielscher** [reply]
You are right, I updated my answer accordingly.

**Comment by rrkamath** [comment]
Thanks Ralf. That worked. 

I had a short follow-up question: What angles do theta and rho denote ? Is theta the azimuthal and rho the tilt ?

**Comment by ralfHielscher** [comment]
It is the other way round.

Ralf.
********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by rrkamath** [reply]
Got it ! Thanks.

Rakesh
